app/src/main/java/com/example/henzoshimada/feeltrip/Mood.java

```diff
@@ -139,48 +152,70 @@ public String getEmotionalState() {
     }
 
     public void setEmotionalState(String emotionalState) {
-        this.emotionalState = emotionalState;
-        setStateByIndex(0);
+        if(!getEmotionalState().equals(emotionalState)) {
+            this.emotionalState = emotionalState;
+            setStateByIndex(0);
+        }
     }
 
     public void setImage(String image) {
-        this.image = image;
-        setStateByIndex(5);
+        if(getImage() == null) {
+            this.image = image;
+            setStateByIndex(5);
+        }
+        else if(!getImage().equals(image)) {
+            this.image = image;
+            setStateByIndex(5);
+        }
```

Comment:
If the image String already matches what we have, there's no point marking the image fields as being "changed"